README.md

```diff
@@ -1,22 +1,130 @@
-Data is retrieved from the `dissco` database and elasticSearch.
+The DiSSCo backend provides APIs which can be used by other applications, such as [DiSSCover](https://sandbox.dissco.tech/).
+It handles the logic and preprocesses the items for serialization.
+In general the find/search APIs are open and APIs for create/update/delete are protected by Oauth (Keycloak).
 
-## API's
-The API's are document with OpenAPI v3 and a Swagger endpoint.
+All endpoints are based on the [JSON:API](https://jsonapi.org/) specification.
+It follows the guidelines and best practices described in [BiCIKL Deliverable 1.3](https://docs.google.com/document/d/1RgngKSPabEs-Pir6vA25iFDgVorbEZe7duT7L7vQ7QI)
+The organisation endpoints form an exception on this.
 
-## Properties
+The backend provides APIs for the following objects:
+- Digital Specimens
+- Digital Media Objects
+- Annotations
+- Users
+- Organisations
 
-### Database
-- `spring.datasource.url` - The JDBC url of the databse
-- `spring.datasource.username` - Username of the databse user with suffecient rights
-- `spring.datasource.password` - Password for the user
+## Storage solutions
+In general, there are three places where data is stored:
+- Postgres database
+The Postgres database stores the latest active version of the object.
+It is used in retrieving a specific object based on the id.
+It can also be used to combine objects based on their relationships.
 
-### ElasticSearch
-- `elasticsearch.hostname` - Hostname of the elasticsearch instance
-- `elasticsearch.port` - Portnumber of the elasticsearch instance
+- Elasticsearch
+This data storage is used for searching and aggregating.
+It is used for data discovery and provides endpoints for the filters and search fields.
+In general, it does not return a single object, but a paginated list of objects.
+Additionally, it can provide aggregations showing how many items comply to the search criteria.
+
+- MongoDB
+MongoDB is used for provenance storage and stores the historical data.
+This data storage is used for displaying previous versions of the data.
+
+## API documentation
+The APIs are documented through code-generated OpenAPI v3 and Swagger.
+- The swagger endpoint is: https://sandbox.dissco.tech/api/swagger-ui/index.html#/
+- The OpenAPI endpoint is: https://sandbox.dissco.tech/api/v3/api-docs
+
+As we use generic objects, the API documentation is not as detailed as we would like.
+We are looking at additional options for more detailed documentation.
+
+### Digital Specimens
+For Digital Specimen, we only provide search and read functionality through the APIs.
+It follows a generic structure.
+`{endpoint}/{prefix}/{suffix}`
+Optionally, this is followed by a version `/{version}` for a specific version.
+It can also be followed by a particular view on the data such as `/full`
+The full endpoint provide all specimen information including all connected information.
```

Comment:
provide's' -> add s at the end of provide

Reply:
:+1: